go-controller/pkg/libovsdbops/acl.go

```diff
@@ -172,10 +173,9 @@ func DeleteACLsOps(nbClient libovsdbclient.Client, ops []libovsdb.Operation,
 		// can't use i in the predicate, for loop replaces it in-memory
 		acl := acls[i]
 		opModel := operationModel{
-			Model:          acl,
-			ModelPredicate: func(item *nbdb.ACL) bool { return isEquivalentACL(item, acl) },
```

Comment:
oh I am trying to understand the commit message, forgive my stupidity. is my understanding correct?

problem, we were doing bulk delete so multiple acls would try to get flushed, if one of them is still being referenced by a PG but others are free to be deleted? so we were failing to delete the free ones as well?

fix: you are removing deletion by predicate totally and not doing bulk ops, so we are not going to actually do the "find and delete" here right? we are simply winging it from now on, just go and try to delete and good luck?

Reply:
Yeah that's right. The main point is we can't delete duplicate ACLs without de-referencing them from port groups. To do that we need ACL.UUIDs, we can either do an additional lookup by predicate in the beginning of this function, or just require for all ACLs to have UUID set.
It could be two different functions (deleting ACLs with UUIDs set and by predicate with lookup), but the only use case for the second option is deleting ACLs together with its port group, and we know it will be deleted in that case anyway, so for performance reasons I didn't add this second option, and only updated existing function.